sdks/java/core/src/main/java/org/apache/beam/sdk/io/ReadAllViaFileBasedSourceTransform.java

```diff
@@ -88,6 +88,7 @@ public SplitIntoRangesFn(long desiredBundleSizeBytes) {
     @ProcessElement
     public void process(ProcessContext c) {
       MatchResult.Metadata metadata = c.element().getMetadata();
+      FileSystems.reportSourceLineage(metadata.resourceId().getCurrentDirectory());
```

Comment:
This is for ReadAll and the challenge here is that we do not know how many files it is going to be, and actually here it can be used in a streaming pipeline where file metadata comes in over time. So do not have hint whether report one level above or file path is preferred (like other places)

- Pro of report one level above:

result still accurate (no entry dropped) when the upstream filepattern is matching, or watching a directory (which is a common use case)

- Con of report one level above:

less granularity, obviously

-----

- Pro of report whole path:

More granularity

- Con of report whole path

More likely to hit metrics limit, and report incomplete list of files


I'm open to both. Personally I prefer the former, the reason is same as the concen in the comments of #32650 , that is, given the limitation of metrics size, I consider the preference is

full information > info with less granularity > has to truncate and report incomplete results